torch/_higher_order_ops/triton_kernel_wrap.py

```diff
@@ -1020,9 +1020,21 @@ def __getitem__(self, *args):
         return tracing_triton_hopifier_singleton.call_getitem(self, args)
 
     def run(self, *args, **kwargs):
-        return tracing_triton_hopifier_singleton.call_run(self, args, kwargs, None)
+        from torch._library.triton import is_capture_triton_enabled
+
+        if is_capture_triton_enabled():
+            return tracing_triton_hopifier_singleton.call_run(self, args, kwargs, None)
+        else:
+            assert self.kernel is not None
+            return self.kernel.run(*args, **kwargs)
```

Comment:
noob q, how is `self.kernel.run(*args, **kwargs)` different from `self.kernel[self.grid](*args, **kwargs)` ? 

without grid, run is `symbolic`?

Reply:
You can invoke a triton kernel in two ways:
1) `kernel[grid](*args, **kwargs)`
2) `kernel.run(*args, **kwargs, grid=<grid>, warmup=<something>)`